File: 5_matematica/1061/1061.py
"""
BEE 1061 - Tempo de um Evento - Nível 3 - Matemática
"""

dia_em_seg = 24 * 60 * 60

dia_inicio = int(input().strip().split(" ")[1])
hora_inicio = [int(i) for i in input().strip().split(" : ")]
dia_termino = int(input().strip().split(" ")[1])
hora_termino = [int(i) for i in input().strip().split(" : ")]


tempo_dia1 = dia_em_seg - (hora_inicio[0] * 3600 + hora_inicio[1] * 60 + hora_inicio[2])
tempo_ultimo_dia = hora_termino[0] * 3600 + hora_termino[1] * 60 + hora_termino[2]

tempo_dia1e2 = tempo_dia1 + tempo_ultimo_dia
w = dia_termino - dia_inicio - 1
w += tempo_dia1e2 // dia_em_seg
x = tempo_dia1e2 % dia_em_seg // 3600
y = tempo_dia1e2 % dia_em_seg % 3600 // 60
z = tempo_dia1e2 % dia_em_seg % 3600 % 60


print(f"{w} dia(s)\n{x} hora(s)\n{y} minuto(s)\n{z} segundo(s)")

File: mathematics/1061/1061.cpp
/*
BEE 1061 - Tempo de um Evento - Level 3 - Mathematics
*/
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int dia_i, dia_t, w, x, y, z, tempo_dia_i, tempo_dia_t, total_tempo;
    string N;

    cin >> N >> dia_i;

    scanf("%d : %d : %d", &x, &y, &z);

    tempo_dia_i = 86400 - (x * 3600 + y * 60 + z);

    cin >> N >> dia_t;

    scanf("%d : %d : %d", &x, &y, &z);
    tempo_dia_t = x * 3600 + y * 60 + z;

    total_tempo = tempo_dia_i + tempo_dia_t;

    w = dia_t - dia_i - 1;
    w += total_tempo / 86400;
    x = total_tempo % 86400 / 3600;
    y = total_tempo % 86400 % 3600 / 60;
    z = total_tempo % 86400 % 3600 % 60;

    cout << w << " dia(s)" << endl;
    cout << x << " hora(s)" << endl;
    cout << y << " minuto(s)" << endl;
    cout << z << " segundo(s)" << endl;

    return 0;
}